curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.md

```diff
@@ -85,6 +99,6 @@ let result = false; // Change this line
 
 ```js
 let myString = "Eleanor Roosevelt";
-let myRegex = /(Franklin|Eleanor).*Roosevelt/;
+let myRegex = /(Franklin|Eleanor)\s{1}(\S+\s)*Roosevelt/;
```

Comment:
This regex would pass `Eleanor ?dsg Roosevelt` as valid.  That is a problem.

How about:
```
/(Franklin|Eleanor) (([A-Z]\.?|[A-Z][a-z]+) )?Roosevelt/
```

Reply:
Your regex is failing in this case
![image](https://user-images.githubusercontent.com/91016903/198698196-c7189e6e-dd79-49ba-9dc8-a59b760e4e9d.png)

So I've modified my regex. Shall I use this ?
`(Franklin|Eleanor)\s([A-Za-z]+\.?\s)*Roosevelt`

Reply:
For your new regex, `Eleanor dsg. asdf Roosevelt` should fail anyway as it is only supposed to allow for a middle name and not two and definitely not `dsg.`  A valid abbreviation would be a single capital letter followed by a period like 'D.`